astropy/table/tests/test_init_table.py

```diff
@@ -10,6 +10,37 @@
 import astropy.units as u
 
 
+class DictLike(Mapping):
+    """A mapping-like object that does not subclass dict.
+
+    This is used to test code that expects dict-like but without actually
+    inheriting from dict.
+    """
+    def __init__(self, *args, **kwargs):
+        self._data = dict(*args, **kwargs)
+
+    def __getitem__(self, item):
+        return self._data[item]
+
+    def __setitem__(self, item, value):
+        self._data[item] = value
+
+    def __iter__(self):
+        return iter(self._data)
+
+    def __len__(self):
+        return len(self._data)
+
+    def items(self):
+        return self._data.items()
+
+    def keys(self):
+        return self._data.keys()
+
+    def values(self):
+        return self._data.values()
```

Comment:
You should need only `__getitem__`, `__iter__` and `__len__`, since `Mapping` will provide the other methods. Not sure about `__setitem__`, maybe some tests are setting values ?

Reply:
Done, but I left __setitem__ just to make this actually work as a minimal dict-like object, but confirmed it is not needed for testing.